File: yts_project_vueai/src/api/modules/referenceProductFamily.ts
import request from '../request'

export interface ReferenceProductFamily {
  id?: number
  category: string
  subCategory?: string
  detailName: string
  displayOrder?: number
  createdAt?: string
  updatedAt?: string
}

/** 获取所有产品家族 */
export function getAllProductFamilies(): Promise<ReferenceProductFamily[]> {
  return request.get('/reference/product-family')
}

/** 根据ID获取 */
export function getProductFamilyById(id: number): Promise<ReferenceProductFamily> {
  return request.get(`/reference/product-family/${id}`)
}

/** 根据大类查询 */
export function getProductFamilyByCategory(category: string): Promise<ReferenceProductFamily[]> {
  return request.get(`/reference/product-family/category/${category}`)
}

/** 获取所有大类列表 */
export function getDistinctCategories(): Promise<string[]> {
  return request.get('/reference/product-family/categories')
}

/** 获取子类列表 */
export function getDistinctSubCategories(category?: string): Promise<string[]> {
  return request.get('/reference/product-family/sub-categories', { params: { category } })
}

/** 创建 */
export function createProductFamily(data: ReferenceProductFamily): Promise<ReferenceProductFamily> {
  return request.post('/reference/product-family', data)
}

/** 更新 */
export function updateProductFamily(id: number, data: ReferenceProductFamily): Promise<ReferenceProductFamily> {
  return request.put(`/reference/product-family/${id}`, data)
}

/** 删除 */
export function deleteProductFamily(id: number): Promise<void> {
  return request.delete(`/reference/product-family/${id}`)
}

File: yts_project/src/main/java/com/it/yts_project/model/ReferenceProductFamily.java
package com.it.yts_project.model;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 产品家族字典表实体类
 * 用于存储产品分类体系：大类 > 子类 > 详细名称
 */
@Data
public class ReferenceProductFamily {

    /**
     * 主键ID
     */
    private Integer id;

    /**
     * 大类：书版/咭书/活页书/盒/拼图等
     */
    private String category;

    /**
     * 子类
     */
    private String subCategory;

    /**
     * 详细名称
     */
    private String detailName;

    /**
     * 显示排序
     */
    private Integer displayOrder;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;
}
